arch/xtensa/soc/esp32/peripheral.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __PERIPHERAL_H__
+#define __PERIPHERAL_H__
+
+#include "soc.h"
+
+static void esp32_enable_peripheral(const struct esp32_peripheral *peripheral)
```

Comment:
Please move this function to `soc.h`, since `struct esp32_peripheral` is defined there.  While at it, also make this `static inline void`, since it's defined in a header file.